### ISSUE TITLE
function K_MostImportant_features

### ISSUE THREAD

**Issue by db-beep** [JohnYeung-dojjy/FYP-scDHA#1]
Hi,author:   
 if k > data.shape[0]:
        k = data.shape[0]
in denoise.py line 63,64.select 5000 important gene,data.shape[0] = 90(in yan dataset),so it select 90 important gene.
can you explain it?
thanks

**Comment by cfyeung-dojjy** [comment]
Hi, the input to the function is a transposed matrix, so its size is [feature_number]*[number of cell]
this is a control to prevent index out of bound.

**Comment by cfyeung-dojjy** [comment]
Actually, I need to check, thanks for pointing it out.

**Comment by db-beep** [comment]
in scDHA.py,there is some reference error,in line 24,27,28(about paper_encode)
import denoise, encode, paper_encode
from encode import stacked_bayesian_autoencoder
from paper_encode import paper_encoder
I think the paper said input to the function is a transposed matrix,so its size is [[number of cell]*feature_number](in scDHA method paper)
I change the code(data.shape[1]),I don't know this is right!
thanks

**Comment by cfyeung-dojjy** [comment]
Sorry that I am still busy with my school work so I won't dig into it for now. You can expect to receive further respond after mid-May.
Meanwhile, you can take a look at the repo by the author of scDHA first and compare their R implementation with this python code.

The code works well on my side, at least it didn't have error running the code.
There could be some name collusion when I merged the branches before pushing here.

BTW, the file paper_encode is dispatched, you can replace it with encode.py instead

**Comment by cfyeung-dojjy** [comment]
for the input to the function, you can print the shape and see if it is correct

**Comment by db-beep** [comment]
thanks
good luck!

**Comment by cfyeung-dojjy** [comment]
Note that array index start with 1 in R, so arr[[1]] in R = arr[0] in python

**Comment by cfyeung-dojjy** [comment]
Hi, sorry for the late reply, I tested the code again, it turns out that I made an mistake, and it should be data.shape[1]. I didn't realize it because even if there were only ~[100-1000] features, the result is still not bad. I will push a new version where this is corrected. Thank you for pointing it out

> in scDHA.py,there is some reference error,in line 24,27,28(about paper_encode) import denoise, encode, paper_encode from encode import stacked_bayesian_autoencoder from paper_encode import paper_encoder I think the paper said input to the function is a transposed matrix,so its size is [[number of cell]*feature_number](in scDHA method paper) I change the code(data.shape[1]),I don't know this is right! thanks

As for the dependency issue, I have handled it and will push the new version shortly

**Comment by db-beep** [comment]
Sorry, I just saw your reply. Because I want to use some data processing methods in scdha, as I said before, data.shape [1] should be set like this. Thank you for your reply and for taking the time to check the code.

**Comment by cfyeung-dojjy** [comment]
yes, thank you for pointing it out. Feel free to use the code, but please note that I didn't implement the optimized function for big data (dataset with >5000 samples).
